blockchain/transaction_confirmers.go

```diff
@@ -102,22 +101,71 @@ func (t *TransactionConfirmer) Complete() bool {
 	return t.complete
 }
 
-// InstantConfirmations is a no-op confirmer as all transactions are instantly mined so no confirmations are needed
-type InstantConfirmations struct{}
+// L2TxConfirmer is a near-instant confirmation method, primarily for optimistic L2s that have near-instant finalization
+type L2TxConfirmer struct {
+	client   EVMClient
+	txHash   common.Hash
+	complete bool
+	context  context.Context
+	cancel   context.CancelFunc
+}
 
-// ReceiveBlock is a no-op
-func (i *InstantConfirmations) ReceiveBlock(block NodeBlock) error {
-	return nil
+func NewL2TxConfirmer(client EVMClient, txHash common.Hash) *L2TxConfirmer {
+	ctx, ctxCancel := context.WithTimeout(context.Background(), client.GetNetworkConfig().Timeout)
+	return &L2TxConfirmer{
+		client:   client,
+		txHash:   txHash,
+		complete: false,
+		context:  ctx,
+		cancel:   ctxCancel,
+	}
 }
 
-// Wait is a no-op
-func (i *InstantConfirmations) Wait() error {
+// ReceiveBlock does a quick check on if the tx is confirmed already
+func (l *L2TxConfirmer) ReceiveBlock(block NodeBlock) error {
+	confirmed, err := l.client.IsTxConfirmed(l.txHash)
+	if err != nil {
+		return err
+	}
+	if confirmed {
+		l.complete = true
+	}
 	return nil
 }
 
+// Wait checks every 50 milliseconds if the tx has been confirmed or not
+func (l *L2TxConfirmer) Wait() error {
+	countdown := time.NewTimer(0)
+	defer func() {
+		l.cancel()
+		countdown.Stop()
+		l.complete = true
+	}()
+	if l.complete {
+		return nil
+	}
+
+	for {
+		select {
+		case <-countdown.C:
+			confirmed, err := l.client.IsTxConfirmed(l.txHash)
```

Comment:
We have confirmations here on ticks and in `ReceiveBlock` one is not enough?

Reply:
Faster this way. On Arbitrum testing, this cut down wait times by ~1s on average.

Reply:
Well, that is really, really strange, because I get that L2 block time is fast, but then they should be confirmed in the next block at least, not before.